src/main/kotlin/lifeCycle/LifeCycle.kt

```diff
@@ -0,0 +1,32 @@
+package lifeCycle
+
+class LifeCycle(state: LifeCycleState) {
+    private var lifeCycleListeners = mutableListOf<LifeCycleListener>()
+
+    var state: LifeCycleState = state
+        private set
+
+    fun changeState(state: LifeCycleState) {
+        this.state = state
+        lifeCycleListeners.forEach { it.onStateChange(state) }
+        if (state == LifeCycleState.DESTROYED)
+            lifeCycleListeners.clear()
+    }
+
+    fun isChangeAble(): Boolean {
```

Comment:
```suggestion
    fun isChangeable(): Boolean {
```
접미사에 따로 케이스분기를 하지는 않습니다 😄